### ISSUE TITLE
Two new optimization techniques: NorMuon and Cautious Weight Decay

### ISSUE THREAD

**Issue by kabachuha** [tdrussell/diffusion-pipe#464]
NorMuon and Cautious Weight Decay have recently set two new records on the GPT speedrun repository and proven effective in their respective accompanying science papers. (with the latter technique exhaustively tested with 20,000 GPU hours)

Both methods work out-of-the-box, with minimal modifications to the Muon/GenericOptim class.

NorMuon: https://arxiv.org/pdf/2510.05491, https://github.com/zichongli5/NorMuon.

Cautious Weight Decay: https://arxiv.org/pdf/2510.12402, https://github.com/kozistr/pytorch_optimizer/commit/99dc4a536181892e4a8f2f60c292345d6dc1b566.

I think they would be much useful for training acceleration / convergence.

**Comment by frauttauteffasu** [comment]
@kabachuha Have you had an opportunity to compare AdaGO https://arxiv.org/abs/2509.02981 with Normuon?
